app/workers/detect_invariants.rb

```diff
@@ -185,6 +186,30 @@ def detect_application_choices_with_courses_from_the_incorrect_cycle
     end
   end
 
+  def detect_submitted_applications_with_an_incorrect_number_of_selected_references
+    applications_with_incorrect_number_of_references = ApplicationForm
+    .joins(:application_references)
+    .where.not(submitted_at: nil)
+    .where(references: { feedback_status: :feedback_provided })
+    .where(references: { selected: true })
```

Comment:
Can remove the `.where(references: { feedback_status: :feedback_provided })` I think, ie—it's sufficient to just query for selected references.